frontend/src/components/samples/details/SampleDetailsContent.js

```diff
@@ -139,40 +112,49 @@ const SampleDetailsContent = ({
 
 
   // Navigate to a sample when the sample's node is clicked in the lineage graph.
-  const navigateToSample = (sample_id) => {
+  const navigateToSample = useCallback((sample_id) => {
     if (sample_id) {
       history(`/samples/${sample_id}#lineage`)
     }
-  }
+  }, [history])
 
   // Navigate to a process measurement when the user clicks an edge
   // in the lineage graph.
-  const navigateToProcess = (process_id) => {
+  const navigateToProcess = useCallback((process_id) => {
     if (process_id) {
       history(`/process-measurements/${process_id}`)
     }
-  }
+  }, [history])
 
   // TODO: This spams API requests
-  if (!samplesByID[id])
-    getSample(id);
+  useEffect(() => {
+    if (!samplesByID[id])
+      dispatch(getSample(id));
+  }, [dispatch, id, samplesByID])
 
-  if (isLoaded && !sample.versions && !sample.isFetching)
-    listVersions(sample.id);
+  useEffect(() => {
+    if (isLoaded && !sample.versions && !sample.isFetching)
+      dispatch(listVersions(sample.id));
+  }, [dispatch, isLoaded, sample.id, sample.isFetching, sample.versions])
 
-  if (isLoaded && !isProcessesEmpty) {
-    sample.process_measurements.forEach((id, i) => {
-      withProcessMeasurement(processMeasurementsByID, id, process => process.id);
-      processMeasurements.push(processMeasurementsByID[id]);
-    })
-  }
+  useEffect(() => {
+    if (isLoaded && !isProcessesEmpty) {
+      fetchProcessMeasurements(sample.process_measurements).then((processMeasurements) => {
+        setProcessMeasurements(processMeasurements)
+      })
+    }
+  }, [isLoaded, isProcessesEmpty, sample.process_measurements])
 
-  if (isLoaded && container?.experiment_run) {
-    experimentRunsIDs.push(container.experiment_run)
-  }
+  useEffect(() => {
+    if (isLoaded && container?.experiment_run) {
+      setExperimentRunIDs((experimentRunsIDs) => [...experimentRunsIDs, container.experiment_run])
```

Comment:
Infinite loop of dead

Reply:
Looks like it's getting run twice ... adding twice the same id to the list ... then when it reload it loads correctly. 
I really don't see how a single sample could be part of more than one experiment. In the current way it is designed it should always have either zero or one experiment run associated ... I would get rid of the array maybe even of the state ...

Reply:
Should be fixed by now.